### DISCUSSION TITLE
using cartesianInterfaceControl

### DISCUSSION THREAD

**Discussion by matejhof** [robotology/community#140] [Q&A] [answered]
With iKinCartesianSolvers running, following the info [here](http://wiki.icub.org/brain/group__cartesianInterfaceControl.html), I'm doing the following: 
`cartesianInterfaceControl --part left_arm --onlyXYZ`
and from another terminal:
`yarp write ... /armCtrl/left_arm/xd:i`
`> (-0.18 -0.1 0.1 1.0 0.0 0.0 0.0)`
(I tried with and without brackets)
But there is no response.
Can someone, most likely @pattacini :), please tell me what is wrong?

**Comment by pattacini** [comment, marked as answer]
@matejhof just tested on my system against `icubSim` and it's perfectly working.
Double check that all the required resources are running fine on the robot / your system.

If you launch the module with `--onlyXYZ`, then you'd need only 3 doubles (without brackets) as input to that port.

**Comment by matejhof** [comment]
Thanks! The 3 doubles instead of 7 did the trick. Perhaps this could be added to the wiki, or a simple example - or maybe it was just me not realizing this automatically. Closing.

**Comment by pattacini** [comment]
Should work also with more than 3 doubles in `--onlyXYZ`; see https://github.com/robotology/icub-main/blob/master/src/modules/cartesianInterfaceControl/main.cpp#L276.
